assignments/this.js

```diff
@@ -1,26 +1,49 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. The first rule of this is when a function is available in the global scope, the value of this 
+    becomes the window object.
+
+* 2. The second rule of this is when a function is called by a preceding dot, this becomes the object before the 
+    preceding dot.
+
+
+* 3. The third principle of this key word is usage of `new` keyword when instantiating a constructor function.
+    this becomes the instance of the function.
+
+* 4. The fourth rule of this is using call(), or apply(), or bind() to explicitly bind this.
 *
 * write out a code example of each explanation above
 */
+function GameObject(name, dimensions, createdAt) {
+    this.name = name;
```

Comment:
Why is this function here? 
Be careful and try to clean up your code before pushing your code.